### PR TITLE
Network Monitor + Storybook

### PR DESCRIPTION
**Pull request by jcmorrow**
This is still a work in progress, but I think it will be ready to go out soon!

CleanShot 2021-11-11 at 02 09 32

### CODE REVIEW

File: src/ui/reducers/network.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const getRequests = (state: UIState) => state.network.requests;
+
+export const selectors = {
+  getEvents,
+  getRequests,
+  getNetworkRequests: createSelector(getEvents, getRequests, (events, requests) => ({
+    events,
+    requests,
+  })),
+};
```
Comment (jasonLaster) [inline src/ui/reducers/network.ts:44]: I think in the past we've exported each selector individually and done `* as selectors` elsewhere. I'm spoiled, but at this point an export object seems like a hassle

File: src/ui/components/shared/MaterialIcon.tsx [hunk 2]
```diff
@@ -29,16 +26,12 @@ export default function MaterialIcon({
...
-  const fontLoading = useSelector((state: UIState) => selectors.getFontLoading(state));
   return (
     <div
       {...rest}
       className={classnames(
         "leading-none",
         className,
-        {
-          invisible: fontLoading,
-        },
```
Comment (jasonLaster) [inline src/ui/components/shared/MaterialIcon.tsx:41]: I think this commit landed separately, but once this 2nd omnibus lands, it'd be nice to get back into a cadence for smaller PRs :)

File: src/ui/components/SecondaryToolbox/index.tsx [hunk 3]
```diff
@@ -83,6 +80,14 @@ function PanelButtons({
           <div className="label">React</div>
         </button>
       )}
+      <button
```
Comment (jasonLaster) [inline src/ui/components/SecondaryToolbox/index.tsx:83]: We should wrap this in a feature flag, even if it is `features.network` from `ui/utils/prefs`

Comment (jasonLaster) [reply src/ui/components/SecondaryToolbox/index.tsx:83]: partly because i dont know if we want to land this without a secondary toolbox for headers and responses

File: src/ui/components/SecondaryToolbox/index.tsx [hunk 4]
```diff
@@ -45,11 +46,7 @@ function PanelButtons({
     setSelectedPanel(panel);
     trackEvent(`toolbox.secondary.${panel}_select`);
 
-    // The comments panel doesn't have to be initialized by the toolbox,
-    // only the console and the inspector.
-    if (panel !== "comments") {
```
Comment (jasonLaster) [inline src/ui/components/SecondaryToolbox/index.tsx:50]: oh funny...

File: src/ui/components/NetworkMonitor/Status.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+const Status = ({ value }: StatusProps) => {
+  const family = familyFor(value);
+  const familyClass = FAMILY_CLASSES[family];
+  return (
+    <div
+      className={classNames(
+        "status font-semibold inline-block rounded-md  p-1",
+        familyClass.map(c => styles[c])
```
Comment (jasonLaster) [inline src/ui/components/NetworkMonitor/Status.tsx:39]: nice

File: src/ui/components/Library/RecordingRow.tsx [hunk 6]
```diff
@@ -53,7 +53,6 @@ function RecordingRow({
   removeSelectedId,
   loadReplayPrefs,
 }: RecordingRowProps) {
-  const router = useRouter();
```
Comment (jasonLaster) [inline src/ui/components/Library/RecordingRow.tsx:56]: 👍

File: src/ui/components/Library/Invitations.tsx [hunk 7]
```diff
@@ -9,7 +9,7 @@ export default function Invitations({}) {
     return null;
   }
 
-  const displayedWorkspaces = [...pendingWorkspaces!].sort((a, b) =>
+  const displayedWorkspaces = [...(pendingWorkspaces || [])].sort((a, b) =>
```
Comment (jasonLaster) [inline src/ui/components/Library/Invitations.tsx:12]: these small fixes are great, but should be in their own PRs that land quickly. Like i think this broke for me recently in dev

File: src/protocol/thread/thread.ts [hunk 8]
```diff
@@ -774,6 +776,24 @@ class _ThreadFront {
...
+      requests,
+      events,
+    }: {
+      requests: RequestInfo[];
+      events: RequestEventInfo[];
+    }) => void
+  ) {
+    const sessionId = await this.waitForSession();
+    client.Network.addRequestsListener(
+      async ({ events, requests }: { requests: RequestInfo[]; events: RequestEventInfo[] }) => {
```
Comment (jasonLaster) [inline src/protocol/thread/thread.ts:790]: nit
```js
   async ({ events, requests }: { requests: RequestInfo[]; events: RequestEventInfo[] }) => 
        onRequestsReceived({ requests, events })
    );
```

File: src/devtools/client/webconsole/actions/network.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { RequestInfo, RequestEventInfo } from "@recordreplay/protocol";
+import { ThreadFront } from "protocol/thread";
+
+export const setupNetwork = (store: any) => {
+  ThreadFront.findNetworkRequests(({ requests, events }) => {
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/actions/network.ts:5]: when we can drop the curlies, it's nice to since it's more deliberate

File: src/devtools/client/themes/variables.css [hunk 10]
```diff
@@ -2,12 +2,6 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-/*
- * IMPORTANT NOTE
- * This file is parsed in js (see client/shared/theme.js)
- * so the formatting should be consistent (i.e. no '}' inside a rule).
- */
```
Comment (jasonLaster) [inline src/devtools/client/themes/variables.css:9]: lol

File: src/base.css [hunk 11]
```diff
@@ -18,22 +18,24 @@ body {
   border: 2px solid black !important;
 }
 
+:root {
```
Comment (jasonLaster) [inline src/base.css:21]: Nice

**Review by jasonLaster — COMMENTED** [review]
Nice! Looking great

File: src/ui/reducers/network.ts (hunk 1 above)
Comment (jcmorrow) [reply src/ui/reducers/network.ts:44]: Yeah, I saw that 😄 . I am ok with sticking with that style for this purpose. In general I stay pretty far away from `*` imports.

File: src/ui/components/shared/MaterialIcon.tsx (hunk 2 above)
Comment (jcmorrow) [reply src/ui/components/shared/MaterialIcon.tsx:41]: Yep, agreed!

File: src/ui/components/SecondaryToolbox/index.tsx (hunk 3 above)
Comment (jcmorrow) [reply src/ui/components/SecondaryToolbox/index.tsx:83]: Yep, I started adding a setting in the same place that we have other experimental features, and that seemed to be a little involved, so I figured I would bring it up tomorrow and see what people think is the best way to do it.

File: src/ui/components/Library/Invitations.tsx (hunk 7 above)
Comment (jcmorrow) [reply src/ui/components/Library/Invitations.tsx:12]: Yep, you're right.

File: src/ui/reducers/network.ts (hunk 1 above)
Comment (jasonLaster) [reply src/ui/reducers/network.ts:38]: haha fair enough

File: src/ui/components/SecondaryToolbox/index.tsx (hunk 3 above)
Comment (jasonLaster) [reply src/ui/components/SecondaryToolbox/index.tsx:83]: yeah - userSetting is a column in the DB which is its own thing... but features.network would be 5 lines...

File: src/ui/components/Library/Invitations.tsx (hunk 7 above)
Comment (jcmorrow) [reply src/ui/components/Library/Invitations.tsx:12]: https://github.com/RecordReplay/devtools/pull/4453/files

**Comment by jcmorrow** [comment]
I'm going to add a header display first thing this morning, but I think this is ready to go out behind the feature flag 🏁

**Comment by jcmorrow** [comment]
Closing in favor of separate PRs: #4455 and a Storybook PR (WIP)
